doc/OnlineDocs/contributed_packages/community.rst

```diff
@@ -0,0 +1,274 @@
+
+The three graphs above are all variable graphs - which means the nodes represent variables in the model, and the edges
+represent constraint equations. The coloring differs because the three graphs rely on community maps that were
+created based on a constraint node graph, a bipartite graph, and a variable node graph (from left to right). For
+example, the community map that was generated from a constraint node graph (`type_of_community_map='c'`) resulted
+in three communities (as seen by the purple, yellow, and blue nodes).
+
+For our final example, we will use `generate_model_graph` - this function can be used to create a NetworkX
+graph for a Pyomo model. Here, we will create a NetworkX graph from the model in our first example and
+then create the edge and adjacency list for the graph.
+
+`generate_mode_graph` returns a NetworkX graph of the given model, a dictionary that maps the numbers used to
+represent the model components to the actual components (because Pyomo components cannot be directly added to
+a NetworkX graph), and a dictionary that maps constraints to the variables in them. For this example, we will
+only need the NetworkX graph of the model and the number-to-component mapping.
+
+    >>> model = decode_model_1()
+    >>>
+    >>> # model_graph is a NetworkX graph of the model, and number_component_map is a dictionary
+    >>> # that maps the numbers used to represent the model components to the actual components
+    >>> model_graph, number_component_map, constr_var_map = generate_model_graph(model, type_of_graph='c')
+
+The next two lines are used to create a mapping to change the node values from numbers into strings and the
+second line uses this mapping to create string_model_graph, which has the relabeled nodes.
+
+    >>> string_map = dict((number, str(comp)) for number, comp in number_component_map.items())
+    >>> string_model_graph = nx.relabel_nodes(model_graph, string_map)
+
+Now, we print the edge list and the adjacency list:
+
+    >>> print('Edge List:')
+    >>> for line in nx.generate_edgelist(string_model_graph):
+    >>>     print(line)
+    >>>
+    >>> print('Adjacency List:')
+    >>> for line in nx.generate_adjlist(string_model_graph):
+    >>>     print(line)
+
+The edge and adjacency lists are shown below; also, it is worth mentioning that in the code above, we do not
+have to create `string_map` to create an edge list or adjacency list, but for the sake of having an
+understandable output, it is quite helpful. (Without relabeling the nodes, the output below would not have the
+strings of the components but instead would have integer values.)
+
+    >>> # Output:
+    >>> Edge List:
+    >>> c1 c2 {'weight': 2}
+    >>> c1 c3 {'weight': 1}
+    >>> c2 c3 {'weight': 1}
+    >>> c3 c4 {'weight': 2}
+    >>> c3 c5 {'weight': 2}
+    >>> c4 c5 {'weight': 2}
+    >>> Adjacency List:
+    >>> c1 c2 c3
+    >>> c2 c3
+    >>> c3 c4 c5
+    >>> c4 c5
+    >>> c5
+
+Main Function List
```

Comment:
I am not sure if it is better to use `.. autoclass:: ` here. An example can be found in [gdpopt.rst](https://github.com/Pyomo/pyomo/blob/master/doc/OnlineDocs/contributed_packages/gdpopt.rst)

Reply:
Might be `.. automodule::` instead here.